modules/internal/DefaultRectangular.chpl

```diff
@@ -494,45 +494,63 @@ module DefaultRectangular {
                       followThis.size:string + "D expression with a " +
                       this.rank:string + "D domain)");
 
-      proc anyStridable(rangeTuple, param i: int = 0) param do
-        return if i == rangeTuple.size-1 then rangeTuple(i).stridable
-               else rangeTuple(i).stridable || anyStridable(rangeTuple, i+1);
-
       if chpl__testParFlag then
         chpl__testPar("default rectangular domain follower invoked on ", followThis);
       if debugDefaultDist then
         chpl_debug_writeln("In domain follower code: Following ", followThis);
 
-      param stridable = this.stridable || anyStridable(followThis);
-      var block: rank*range(idxType=intIdxType, stridable=stridable);
+      param newStrides = chpl_strideProduct(this.strides,
+                                            chpl_strideUnion(followThis));
+
+      var block: rank*range(idxType=intIdxType, strides=newStrides);
       if boundsChecking then
         for param i in 0..rank-1 do
           if followThis(i).highBound >= ranges(i).sizeAs(uint) then
             HaltWrappers.boundsCheckHalt("size mismatch in zippered iteration (dimension " + i:string + ")");
-      if stridable {
+
+      if ! newStrides.isPosNegOne() {
         type strType = chpl__signedType(intIdxType);
         for param i in 0..rank-1 {
-          // See domain follower for comments about this
           const rStride = ranges(i).stride;
           const rSignedStride = rStride:strType,
                 fSignedStride = followThis(i).stride:strType;
-          if rStride > 0 {
+          if ranges(i).hasPositiveStride() {
```

Comment:
`rStride.isPositive()`?

Reply:
I use `range.hasPositiveStride()` because it is a param in some cases, whereas `rStride.isPositive()` cannot be a param because `rStride` is an integer and not a param.